src/3_branch.js

```diff
@@ -286,6 +286,15 @@ Branch.prototype['init'] = wrap(callback_params.CALLBACK_ERR_DATA, function(done
 	link_identifier = WEB_BUILD ? (utils.getParamValue('_branch_match_id') || utils.hashValue('r')) : (url ? utils.getParamValue(url) : null),
 	freshInstall = !sessionData || !sessionData['identity_id'],
 
+	check_has_app = function(data, cb) {
+		if (!sessionData) { sessionData = session.get(self._storage); }
```

Comment:
I would be hesitant of using this outer scope's sessionData, since another function could have changed it in the mean time. Let's do the sesion.get every time, and use a `var` so it gets scoped only to this function